#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/CircleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  basicTimeStep 12
  contactProperties [
    ContactProperties {
      material1 "pill"
      coulombFriction [
        0
      ]
      bounce 0.2
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
Viewpoint {
  orientation -0.17921193849689726 -0.05831622671913137 0.9820805968969284 3.5094461886342625
  position 47.514302423127376 24.606160231271545 34.486292128635505
}
TexturedBackground {
  texture "noon_sunny_empty"
}
TexturedBackgroundLight {
}
CircleArena {
  radius 20
  floorAppearance Parquetry {
    type "dark strip"
  }
}
Robot {
  translation -0.1071786605017715 -0.3869679827611363 7.909042681704227
  rotation 0.0017485402009863144 0.00015810603679987126 0.9999984588036357 0.0407624287680369
  children [
    DEF ALFREDO Group {
      children [
        Speaker {
          name "altavoz"
        }
        Robot {
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct18"
                  maxTorque 200
                }
              ]
              endPoint Solid {
                translation 0 0 0.06
                rotation 0 0 1 1.8000000000000003
                children [
                  DEF ROTOR Shape {
                    appearance PBRAppearance {
                      baseColor 1 0 0
                      metalness 0
                    }
                    geometry Mesh {
                      url [
                        "../meshes/rotor.obj"
                      ]
                    }
                  }
                ]
                name "R3"
                boundingObject USE ROTOR
                physics DEF M3D Physics {
                  density -1
                  mass 2
                  damping Damping {
                  }
                }
              }
            }
          ]
          name "R18"
          controller "rotor_universal"
        }
        Robot {
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct15"
                  maxTorque 200
                }
              ]
              endPoint Solid {
                translation 0 0 3.98
                rotation 0 0 1 1.8000000000000003
                children [
                  DEF ROTOR Shape {
                    appearance PBRAppearance {
                      baseColor 1 0 0
                      metalness 0
                    }
                    geometry Mesh {
                      url [
                        "../meshes/rotor.obj"
                      ]
                    }
                  }
                ]
                name "R2"
                boundingObject USE ROTOR
                physics DEF M3D Physics {
                  density -1
                  mass 2
                  damping Damping {
                  }
                }
              }
            }
          ]
          name "R15"
          controller "rotor_universal"
        }
        Robot {
          translation 0 0 10.44
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct12"
                  maxTorque 200
                }
              ]
              endPoint Solid {
                translation -0.0737387 0.291881 -2.348
                rotation 1.4199699788436549e-06 -0.00017261599742816842 0.9999999851008505 1.79794
                children [
                  DEF ROTOR Shape {
                    appearance PBRAppearance {
                      baseColor 1 0 0
                      metalness 0
                    }
                    geometry Mesh {
                      url [
                        "../meshes/rotor.obj"
                      ]
                    }
                  }
                ]
                name "R1"
                boundingObject USE ROTOR
                physics DEF M3D Physics {
                  density -1
                  mass 2
                  damping Damping {
                  }
                }
              }
            }
          ]
          name "R12"
          controller "rotor_universal"
        }
        Solid {
          translation 0 0 10.14
          children [
            DEF C12 Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Mesh {
                url [
                  "../meshes/contenedores_12.obj"
                ]
              }
            }
          ]
          name "C12"
          boundingObject USE C12
          physics DEF M3D Physics {
            density -1
            mass 3
            damping Damping {
            }
          }
        }
        Robot {
          translation 0.3 -0.08 11.27
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
                anchor -0.33 -0.0999998 -2.13
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct12"
                }
              ]
              endPoint Solid {
                translation -0.3 0.08000000000000002 -2.13
                rotation 0 0 1 0.22999999999999893
                children [
                  DEF ASPA_SERVO Shape {
                    appearance PBRAppearance {
                      roughness 1
                      metalness 0
                    }
                    geometry Box {
                      size 0.75 3.5 0.3
                    }
                  }
                ]
                name "solid(1)"
                physics Physics {
                  density -1
                  mass 0.1
                }
              }
            }
            Solid {
              translation -0.0043988 0.109912 -1.38
              children [
                DEF SERVO Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 2.4 1.8 1.2
                  }
                }
              ]
            }
          ]
          name "S12"
          controller "rotor_universal"
        }
        Solid {
          translation 0 0 6.03
          children [
            DEF C15 Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Mesh {
                url [
                  "../meshes/contenedores_15.obj"
                ]
              }
            }
          ]
          name "C15"
          boundingObject USE C15
          physics USE M3D
        }
        Robot {
          translation 0.3 -0.08 7.15
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
                anchor -0.33 -0.0999998 -2.13
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct15"
                }
              ]
              endPoint Solid {
                translation -0.3 0.07999999999999996 -2.13
                rotation 0 0 1 0.22999999999999893
                children [
                  DEF ASPA_SERVO Shape {
                    appearance PBRAppearance {
                      roughness 1
                      metalness 0
                    }
                    geometry Box {
                      size 0.75 3.5 0.3
                    }
                  }
                ]
                name "solid(1)"
                physics Physics {
                  density -1
                  mass 0.1
                }
              }
            }
            Solid {
              translation -0.0043988 0.109912 -1.38
              rotation 0 0 1 0.04
              children [
                DEF SERVO Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                    transparency 0.05
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 2.4 1.8 1.2
                  }
                }
              ]
            }
          ]
          name "S15"
          controller "rotor_universal"
        }
        Solid {
          translation 0.0160313 0.0764827 2.12851
          rotation 0.01899286689548021 -0.009673869771031584 0.9997728178195002 0.0008207718687865466
          children [
            DEF C18 Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Mesh {
                url [
                  "../meshes/contenedores_18.obj"
                ]
              }
            }
          ]
          name "C18"
          boundingObject USE C18
          physics USE M3D
        }
        Robot {
          translation 0.3 -0.08 3.24
          children [
            Speaker {
              name "altavoz"
            }
            Receiver {
              channel 1
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
                anchor -0.33 -0.0999998 -2.13
              }
              device [
                RotationalMotor {
                  name "aspa_motor_ct18"
                }
              ]
              endPoint Solid {
                translation -0.3 0.07999999999999996 -2.13
                rotation 0 0 1 0.22999999999999893
                children [
                  DEF ASPA_SERVO Shape {
                    appearance PBRAppearance {
                      roughness 1
                      metalness 0
                    }
                    geometry Box {
                      size 0.75 3.5 0.3
                    }
                  }
                ]
                name "solid(1)"
                physics Physics {
                  density -1
                  mass 0.1
                }
              }
            }
            Solid {
              translation -0.0043988 0.109912 -1.38
              rotation 0 0 1 0.04
              children [
                DEF SERVO Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                    transparency 0.05
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 2.4 1.8 1.2
                  }
                }
              ]
            }
          ]
          name "S18"
          controller "rotor_universal"
        }
        Solid {
          translation 0 -0.0500198 2.06
          rotation 1 0 0 1.5708
          children [
            DEF BODY Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0.9
                transparency 0.9
                roughness 0.1
                metalness 0
                IBLStrength 5
              }
              geometry DEF BODY Mesh {
                url [
                  "../meshes/alfredo_body_test4.obj"
                ]
              }
            }
          ]
          name "cuerpo"
          boundingObject USE BODY
          physics USE M3D
        }
        Solid {
          translation 0 -7.27313e-06 1.98
          rotation 1 0 0 1.5708
          children [
            TouchSensor {
              name "sensor_plato"
              boundingObject DEF PLATE Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  transparency 0.1
                  metalness 0
                }
                geometry Mesh {
                  url [
                    "../meshes/plato.obj"
                  ]
                }
              }
            }
            DEF PLATE Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                transparency 0.1
                metalness 0
              }
              geometry Mesh {
                url [
                  "../meshes/plato.obj"
                ]
              }
            }
          ]
          name "plato"
          boundingObject USE PLATE
          physics USE M3D
        }
      ]
    }
  ]
  name "alfredo"
  physics USE M3D
  controller "supervisor_pastilla"
  supervisor TRUE
}
DEF PILL Solid {
  translation 1.39687 3.16998 20.0407
  rotation 0.8769600572075846 0.2079298506640833 -0.4332507764162484 -1.871942651000185
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation 1.77627 3.81399 21.1874
  rotation 0.4369308967595768 0.2920689309883548 -0.8507567989785281 -3.041625307179586
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(1)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation 2.53868 4.9548 16.4751
  rotation -0.4169639534126201 0.2217419752247705 -0.8814599015144907 2.73682
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 1 0.333333 0
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(2)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation 2.77 7.12114 12.5299
  rotation 0.4369310243733241 0.2920680162924307 -0.8507570474577819 -3.041625307179586
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0.333333 0.498039
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(3)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation 3.48047 6.04859 12.6686
  rotation -0.2909910359883186 0.4201880519667605 0.8595151063005373 2.43946
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 0.666667 1 0
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(4)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation 3.85586 6.8206 12.4088
  rotation -0.38110981436431296 0.46835077186990726 0.7971216117281359 3.1373
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 1 0.666667 1
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(5)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
DEF PILL Solid {
  translation -2.43806 3.59987 20.425
  rotation 0.5928729461143732 0.4360569603672207 -0.6770199384663376 -2.1315553071795863
  children [
    DEF PILL Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Capsule {
        height 0.7
        radius 0.3
      }
    }
  ]
  name "pill-1(6)"
  contactMaterial "pill"
  boundingObject USE PILL
  physics Physics {
    density -1
    mass 0.5
    damping Damping {
    }
  }
}
Robot {
  translation 17.22 0 20.11
  children [
    Emitter {
      channel 1
    }
    Shape {
      appearance PBRAppearance {
        metalness 0
      }
    }
  ]
  name "supervisor"
  controller "supervisor_torre"
  supervisor TRUE
}
